test/Window/Joystick.test.cpp

```diff
@@ -0,0 +1,76 @@
+#include <SFML/Window/Joystick.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators_range.hpp>
+
+TEST_CASE("[Window] sf::Joystick")
+{
+    SECTION("Constants")
+    {
+        STATIC_CHECK(sf::Joystick::Count == 8);
+        STATIC_CHECK(sf::Joystick::ButtonCount == 32);
+        STATIC_CHECK(sf::Joystick::AxisCount == 8);
+    }
+
+    SECTION("Identification")
+    {
+        const sf::Joystick::Identification identification;
+        CHECK(identification.name == "No Joystick");
+        CHECK(identification.vendorId == 0);
+        CHECK(identification.productId == 0);
+    }
+
+    // By avoiding calling sf::Joystick::update() we can guarantee that
+    // no joysticks will be detected. This is how we can ensure these
+    // tests are portable and reliable.
+
+    const auto joystick = GENERATE(range(0u, sf::Joystick::Count - 1));
```

Comment:
`GENERATE` is a Catch2 macro that, in this case, generates a range of values from 0 to 7 (inclusive) so that we can test various joystick index inputs without having to write `CHECK` 8 times more.